Observer/Customer.php

```diff
@@ -0,0 +1,171 @@
+            'name' => $customer->getFirstname() . ' ' . $customer->getLastname()
+        ];
+
+        $regions = $customer->getTjRegions();
+
+        if (!empty($regions)) {
+            $customerRegions = [];
+            foreach (explode(',', $regions) as $region) {
+                $customerRegions[] = ['country' => 'US', 'state' => $this->region->load($region)->getCode()];
+            }
+            $data['exempt_regions'] = $customerRegions;
+        }
+
+        if ($customerAddress) {
+            $data += [
+                'country' => $customerAddress->getCountry(),
+                'state' => $customerAddress->getRegionCode(),
+                'zip' => $customerAddress->getPostcode(),
+                'city' => $customerAddress->getCity(),
+                'street' => $customerAddress->getStreetFull()
+            ];
+        }
+
+        if ($event->getName() == 'customer_save_after_data_object' && empty($customer->getTjLastSync())) {
+            try {
+                $response = $this->client->postResource('customers', $data);  //create a new customer
+            } catch (LocalizedException $e) {
+                $message = json_decode($e->getMessage());
+
+                if (isset($message->status) && $message->status == 422) {  //unprocessable
+                    try {
+                        $response = $this->client->putResource('customers', $customer->getId(), $data);
+                    } catch (LocalizedException $e) {
+                        $this->logger->log('ERROR PUT customerId ' . $customer->getId() . " " . $e->getMessage(),
+                            'customers');
+                    }
+                }
+            }
+        } elseif ($event->getName() == 'customer_save_after_data_object') {
+            try {
+                $response = $this->client->putResource('customers', $customer->getId(), $data);
+            } catch (LocalizedException $e) {
+                $message = json_decode($e->getMessage());
+
+                if (isset($message->status) && $message->status == 404) {  //unprocessable
+                    try {
+                        $response = $this->client->postResource('customers', $data);
+                    } catch (LocalizedException $e) {
+                        $this->logger->log('ERROR POST customerId ' . $customer->getId() . " " . $e->getMessage(),
+                            'customers');
+                    }
+                }
+            }
+        }
+
+        if ($event->getName() == 'customer_delete_before') {
+            try {
+                $response = $this->client->deleteResource('customers', $customer->getId());  //delete customer
+            } catch (LocalizedException $e) {
+                $this->logger->log('ERROR DELETE customerId ' . $customer->getId() . "" . $e->getMessage(),
+                    'customers');
+            }
+        } else {
+            $customer->setData('tj_last_sync', $this->date->timestamp());
```

Comment:
If we hit an exception during a `customer_save_after_data_object` event and the fallback request doesn't work or isn't applicable (not `422` or `404`), do we still want to set `tj_last_sync`?